### ISSUE TITLE
Make universal pin labelling

### ISSUE THREAD

**Issue by cleveHEX** [RoboticsBrno/ArduinoLearningKitStarter-library#5]
Label ESP32 and AVR pins so they can be accessed universally (Eg. D1, A1).

- [x] Digital pins
- [x] Analog pins
- [x] Specify common pins for eg. I2C, SPI - needs resolve of RoboticsBrno/ArduinoLearningKitStarter#21 first
- [x] Test everything on a real board SKIPPED - ups, but should work
- [x] Merge Arduino ESP32 espressif/arduino-esp32#1643
- [x] Merge PIO ESP32 platformio/platform-espressif32#113

**Comment by cleveHEX** [comment]
Ax are not possible to map on ESP32, because it already defines "ESP32 Dev Module".

**Comment by yaqwsx** [comment]
What is the problem exactly?

**Comment by JarekParal** [comment]
There is already `#define A1 pinX` probably. Is it true @CornyjK?

And what about create own board configuration for ALKS (for Arduino/PlatformIO)? 

`ESP32 ALKS ...`

**Comment by cleveHEX** [comment]
@JarekParal 
Adding new board to boards.txt and making it's /variants/ file would work, it would also eliminate requirement of this library (at least pin definitions) for ESP32. Or you meant another way? Maybie they would accept pull request with our board variant.
EDIT:
Yes, it's already defined in /variants/ESP32/pins_arduino.h.
Switching to "Electronic SweetPeas - ESP320" would solve this issue, but it does not give ability to change certain things (eg. flash mode and flash size).

**Comment by JarekParal** [comment]
It would be nice to have own board.txt (or pins_arduino.h) and send pull request to Arduino and PlatformIO.

**Comment by JarekParal** [comment]
What we will do with this @CornyjK :-)?

Are the digital pins (D1, D2, ...) on ESP32 implemented?

**Comment by cleveHEX** [comment]
@JarekParal 
[In this PR](https://github.com/RoboticsBrno/ArduinoLearningKitStarter-library/pull/7)

**Comment by JarekParal** [comment]
Ok, could you send pull-request @CornyjK?

**Comment by cleveHEX** [comment]
@JarekParal 
I think it should be completed first (third point above, also second - send pull request to ESP32 library). In my opinion this should be in next release.

**Comment by JarekParal** [comment]
I agree with your answer @CornyjK. Could you define pins in this library for SPI, I2C on the ESP32 as I mentioned in the issue https://github.com/RoboticsBrno/ArduinoLearningKitStarter/issues/21.

**Comment by cleveHEX** [comment]
[Here](https://gist.github.com/CornyjK/e1ccd2652432f40611512bdc72f1bff3) is preview of file for PR.

**Comment by cleveHEX** [comment]
SPI/IIC has the same issue as the analog pins.

**Comment by JarekParal** [comment]
You can set the pins for I2C and SPI on ESP32 in this [functions](https://github.com/RoboticsBrno/ArduinoLearningKitStarter-library/commit/16aab29) as `begin()` parameters: 
- https://github.com/espressif/arduino-esp32/blob/master/libraries/Wire/src/Wire.h#L55
- https://github.com/espressif/arduino-esp32/blob/master/libraries/SPI/src/SPI.h#L55
